### DISCUSSION TITLE
body option is ignored inside Http::withOptions()

### DISCUSSION THREAD

**Discussion by caiokawasaki** [laravel/framework#33301] [General]
- Laravel Version: ^7.0
- PHP Version: 7.4.4
- Database Driver & Version: -

### Description:
I need to make an api call, where I send a file directly in the body of the request, I can make the call normally using `Guzzle`, but when I try to use the `HTTP Client` the api returns an error saying that it was unable to read the contents of the request.

### Steps To Reproduce:

When I make the request using the `Guzzle` client the request works correctly:

```php
public function submitFile(Request $request)
{
    $photo = Storage::put('avatars', $request->file('photo'));
    
    $client = new Client(['verify' => false]);

    try {
        $client->post('api/settings/profile/photo', [
            'body' => Storage::get($photo),
            'headers' => [
                'Content-Type' => Storage::mimeType($photo)
            ]
        ]);
    } catch (ClientException $e) {
    }
}
```

However, when I try to upload the file using the `HTTP Client`, the request does not work and returns an error:

```php
public function submitFile(Request $request)
{
    $photo = Storage::put('avatars', $request->file('photo'));

    Http::withoutVerifying()
        ->withOptions([
            'body' => Storage::get($photo)
        ])
        ->withHeaders([
            'Content-Type' => Storage::mimeType($photo)
        ])
        ->post('api/settings/profile/photo');
}
```

I can't send the `body` attribute inside the `->post('api/settings/profile/photo', $data)` method, because `$data` only accepts an array.

Apparently the `HTTP Client` ignores the `body` option inside `withOptions()`. Does this really occur or am I doing something wrong? I've been trying to make this work for hours, I decided to make the request using `Guzzle` just to be sure and everything worked correctly.

This is the api I'm calling: [https://developers.facebook.com/docs/whatsapp/api/settings/profile#setphoto](https://developers.facebook.com/docs/whatsapp/api/settings/profile#setphoto)

```curl
POST /v1/settings/profile/photo
Content-Type: image/jpeg, or other appropriate type

your-binary-image-content
```

**Comment by driesvints** [comment]
I've taken a look but I don't think we already support binary file requests. I think we'd need a new method for this to work.

You do have (See docs):

```php
$response = Http::attach(
    'attachment', file_get_contents($photo), 'photo.jpg', ['Content-Type' => Storage::mimeType($photo)]
)->post('api/settings/profile/photo');
```

But I'm guessing that's not exactly what you need?

**Comment by caiokawasaki** [comment]
I tried to use this attachment method, but without success, I really need to send the binary in the request body.

If it were possible to accept the body within the options it would already solve the problem.

**Comment by taylorotwell** [comment]
PRs welcome.

**Comment by driesvints** [comment]
Moving to discussions as this is more a feature request.
